src/components/search/SearchDirective.js

```diff
@@ -1,756 +1,259 @@
-                    moveTo(map, zoom, center);
-                    // Make sure the above origins are visible at all
-                    // zoom levels
-                    gaMarkerOverlay.add(map, center, extent, true);
-                  } else {
-                    zoomToExtent(map, extent);
-                    gaMarkerOverlay.add(map, center, extent);
-                  }
-                } else {
-                  unregisterMove();
-                }
-                if (origin === 'feature') {
-                  var layerId = datum.attrs.layer;
-                  var featureId = datum.attrs.featureId;
-                  selectFeature(layerId, featureId);
-                }
-                if (origin === 'layer') {
-                  scope.addLayer(datum.attrs.layer, false);
-                }
-                //Take care of swisssearch parameter
-                var sP = gaPermalink.getParams().swisssearch;
-                if (angular.isDefined(sP) &&
-                    sP.length > 0) {
-                  //If we end up here because swisssearch cause a single
-                  //click, we remove the swisssearch parameter on
-                  //move/zoom. Otherwise, we always remove it
-                  if (gaSwisssearch.singleResult()) {
-                    //attach to property change to remove
-                    var unregProp = map.getView().on('propertychange',
-                                                     function() {
-                      gaPermalink.deleteParam('swisssearch');
-                      map.getView().unByKey(unregProp);
-                    });
-                  } else {
-                    gaPermalink.deleteParam('swisssearch');
-                  }
-                }
-              });
+      $rootScope.$on('$translateChangeEnd', function() {
+        $scope.lang = $translate.use();
+      });
 
-            scope.searchableLayersFilter = gaLayerFilters.searchable;
-
-            scope.$watchCollection('layers | filter:searchableLayersFilter',
-                function(layers) {
-              //TODO: this isn't updated when layers param (like 'time') changes
-              var layerBodIds = [];
-              var timeEnabled = [];
-              var timeStamps = [];
-              angular.forEach(layers, function(layer) {
-                var ts = '';
-                if (layer.time && layer.time.substr(0, 4) != '9999') {
-                  ts = layer.time.substr(0, 4);
+      $scope.layers = $scope.map.getLayers().getArray();
+
+      //Init swisssearch parameter handling
+      var searchParam = gaPermalink.getParams().swisssearch;
+      if (angular.isDefined(searchParam) &&
+          searchParam.length > 0) {
+        var unregister = $scope.$on('gaLayersChange', function() {
+          // At this point layers are not added to the map yet
+          var unregisterLayers = $scope.$watchCollection('layers', function() {
+            $timeout(function() {
```

Comment:
What type of message would I listen to? Can I be sure that all permalink layers would be loaded when I get the message? Within the angular framework, I'm assured to get the message when all layers of the permalink are added. Effectively, I'm doing a 'once' here...I agree that code is maybe more bloated than it could be.

(Speaking theoretically here as I didn't try it out).